### PR TITLE
feat(3418): display subnet info for namespace in emerald cluster

### PR DESCRIPTION
**Pull request by golebu2020**

### CODE REVIEW

File: app/services/backend/private-cloud/products.ts [hunk 1]
```diff
@@ -147,6 +147,13 @@ export async function getPodUsageMetrics(licencePlate: string, environment: stri
   return response.data;
 }
 
+export async function getSubnetForEmerald(licencePlate: string, environment: string, cluster: string) {
+  const response = await instance.get(
```
Comment (junminahn) [inline app/services/backend/private-cloud/products.ts:151]: let's set the type of the response data like `instance.get<string>`.

Comment (golebu2020) [reply app/services/backend/private-cloud/products.ts]: @junminahn the response type has been set.

File: app/app/api/private-cloud/products/[licencePlate]/namespace-subnet/route.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { GlobalRole } from '@/constants';
+import createApiHandler from '@/core/api-handler';
+import { OkResponse, UnauthorizedResponse } from '@/core/responses';
+import { models } from '@/services/db';
+import { getSubnet } from '@/services/k8s';
+import { getPathParamSchema } from '../schema';
+
+const queryParamSchema = z.object({
+  environment: string(),
+  cluster: z.nativeEnum(Cluster),
```
Comment (junminahn) [inline app/app/api/private-cloud/products/[licencePlate]/namespace-subnet/route.ts:12]: let's remove `cluster` from the query params.

File: app/app/api/private-cloud/products/[licencePlate]/namespace-subnet/route.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const { environment } = queryParams;
+  const { cluster } = queryParams;
+  const { licencePlate } = pathParams;
+  const { data: product } = await models.privateCloudProduct.get({ where: { licencePlate } }, session);
+
+  if (!product?._permissions.view) {
+    return UnauthorizedResponse();
+  }
+
+  const subnetInfo = await getSubnet(licencePlate, environment, cluster);
```
Comment (junminahn) [inline app/app/api/private-cloud/products/[licencePlate]/namespace-subnet/route.ts:30]: let's replace `cluster` with `product.cluster`.

File: app/services/backend/private-cloud/products.ts [hunk 4]
```diff
@@ -147,6 +148,13 @@ export async function getPodUsageMetrics(licencePlate: string, environment: stri
   return response.data;
 }
 
+export async function getSubnetForEmerald(licencePlate: string, environment: string, cluster: string) {
```
Comment (junminahn) [inline app/services/backend/private-cloud/products.ts:151]: 1. let's set default value for `enviroment` to `dev` here.
2. let's remove `cluster` from the arguments.

File: app/app/private-cloud/products/(product)/[licencePlate]/layout.tsx [hunk 5]
```diff
@@ -35,7 +36,7 @@ export default privateCloudProductLayout(({ getPathParams, session, children })
 
   const [privateState, privateSnap] = usePrivateProductState();
   const { licencePlate = '' } = pathParams ?? {};
-
+  const environment = 'dev';
```
Comment (junminahn) [inline app/app/private-cloud/products/(product)/[licencePlate]/layout.tsx:39]: let's remove this.

File: app/app/private-cloud/products/(product)/[licencePlate]/layout.tsx [hunk 6]
```diff
@@ -51,6 +52,12 @@ export default privateCloudProductLayout(({ getPathParams, session, children })
     privateState.licencePlate = licencePlate;
   }, [licencePlate]);
 
+  const { data: subnetInfo, isLoading: isLoadingSubnetInfo } = useQuery({
+    queryKey: [licencePlate, currentProduct?.cluster],
+    queryFn: () => getSubnetForEmerald(licencePlate, environment, currentProduct?.cluster || ''),
```
Comment (junminahn) [inline app/app/private-cloud/products/(product)/[licencePlate]/layout.tsx:57]: to
` getSubnetForEmerald(licencePlate)`

File: app/app/private-cloud/products/(product)/[licencePlate]/layout.tsx [hunk 7]
```diff
@@ -51,6 +52,12 @@ export default privateCloudProductLayout(({ getPathParams, session, children })
     privateState.licencePlate = licencePlate;
   }, [licencePlate]);
 
+  const { data: subnetInfo, isLoading: isLoadingSubnetInfo } = useQuery({
+    queryKey: [licencePlate, currentProduct?.cluster],
```
Comment (junminahn) [inline app/app/private-cloud/products/(product)/[licencePlate]/layout.tsx:56]: let's remove `currentProduct?.cluster` from the keys.

**Review by junminahn — APPROVED** [review]

**Pull request merged**